pyiron_atomistics/lammps/base.py

```diff
@@ -371,19 +372,12 @@ def enable_h5md(self):
             "all h5md ${dumptime} dump.h5 position force create_group yes"
         )
 
-    def write_input(self):
-        """
-        Call routines that generate the code specific input files
-
-        Returns:
-
-        """
+    def get_input_file_dict(self):
```

Comment:
Nevertheless I want a doc string here, which indicates that it's part of the new routine even though inside lammps base it is not obvious.

Reply:
Added the Docstring

Reply:
Actually what I wanted was more something like `executed by GenericJob.write_input`, because this kind of functions which belong to the workflow routines are extremely difficult to follow when you just look at the child classes.

Reply:
As we discussed on Monday, the process of converting the job model in pyiron from an object model to a functional approach still takes multiple pull requests, the change to more dictionary based data types is one of them. The second is the introduction of calculate functions, which take these input dictionaries as input, call the executable and return the output as dictionary. For the `ExecutorContainerJob` I created a first draft of this change in https://github.com/pyiron/pyiron_base/pull/1477 . This allows pyiron to directly use the executor infrastructure from `pympipool` not only to execute serial python processes but also call executables. So from the `get_input_file_dict()` function we get the input for the `calculate()` function and we need these `get_input_file_dict()` functions to give expert users the option to directly manipulate the input files before they are written.

Reply:
Wherever it is used, it is just extremely difficult to figure out for new developers which functions belong to the workflow routine and which ones not. So it's not so much about what this function does, but it's more about why this function is there. It can also be a comment in the code.

Reply:
Yes, I agree that during the migration from the previous object based execution where we always transferred the job object from one function to the next to the new more functional approach we have redundancies and the code it harder to read. Still I think this is a necessary migration as we can leverage the whole infrastructure already developed for the submission of functions in the `concurrent.futures.Executor` classes and more recently in `pympipool` as we discussed in the pyiron meeting this Monday.

Reply:
But I'm not saying you shouldn't do it. I'm saying there should be one line of comment in the code saying why this function exists.